packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -134,7 +134,10 @@ export function usePasteHandler( props ) {
 				tagName,
 			} );
 
-			if ( typeof content === 'string' ) {
+			if (
+				typeof content === 'string' ||
+				content instanceof RichTextData
```

Comment:
would it make sense to provide a helper for this, like `RichTextData.isRichText` that performs this exact check? that would prevent the need from every call-site to remember that it could be text or structure

Reply:
I'm not sure. We'd also not want to consider a string rich text. This is just here for compatibility and there's probably few places other than internally that need this check.

Reply:
Actually, you are quite right here. I want to check where exactly the rich text data is coming from because normally `content` coming from `pasteHandler` should keep returning a string.

Reply:
Removed it, it was unnecessary. A remnant from an earlier version.